src/utils/positionpnl/positionpnl.js

```diff
@@ -0,0 +1,66 @@
+import {
+  retrieveData,
+  insertData,
+  putData
+} from '../network'
+import PositionPNL from '../../transactions/PostionPNL/PositionPNL'
+
+export function retrieve({ AMId }, callback) {
+  const params = {
+    AMaaSClass: 'positionpnl',
+    AMId
+  }
+  let promise = retrieveData(params)
+    .then(result => {
+      result = result.map(positionpnl => _parsePositionPNL(pnl))
+      if (typeof callback === 'function') {
+        callback(null, result)
+      }
+      return result
+    })
+    if (typeof callback !== 'function') {
+      return promise
+    }
+}
+
+export function amend({ AMId, data }, callback) {
+  const params = {
+    AMaaSClass: 'positionpnl',
+    data: PositionPNL,
```

Comment:
You need to either pass the raw `data` or pass `new PositionPNL(data)`, otherwise `data` doesn't go anywhere.